### DISCUSSION TITLE
[New Code Suggestion] Follow PEP standards for method & variable names

### DISCUSSION THREAD

**Discussion by softwareengineerprogrammer** [NatLabRockies/GEOPHIRES-X#82] [General]
PEP prescribes `snake_case` for function/variable names rather than `CamelCase`. Example in, https://github.com/NREL/GEOPHIRES-X/blob/a18cd171ffeef01de93ccb46c96a00673ac582d3/src/geophires_x/WellBores.py#L242 

should be:

```python
def inj_pressure_drops_and_pumping_power_using_impedence_model(f1: float, v_inj: float, rho_water_inj: float, depth: float,
                                                       well_flow_rate: float, inj_well_diam: float, ninj: int,
                                                       water_loss: float, pump_eff: float, dp_overall) -> tuple:

```

**Comment by malcolm-dsider** [comment]
This is good advice I will adopt going forward as I refactor code.

 

From: Jonathan Pezzino ***@***.***> 
Sent: Monday, December 18, 2023 11:49 AM
To: NREL/GEOPHIRES-X ***@***.***>
Cc: Malcolm Ross ***@***.***>; Assign ***@***.***>
Subject: [NREL/GEOPHIRES-X] [New Code Suggestion] Follow PEP standards for method & variable names (Issue #81)

 

PEP prescribes snake_case for function/variable names rather than CamelCase. Example in, https://github.com/NREL/GEOPHIRES-X/blob/a18cd171ffeef01de93ccb46c96a00673ac582d3/src/geophires_x/WellBores.py#L242 InjPressureDropsAndPumpingPowerUsingImpedenceModel should be named inj_pressure_drops_and_pumping_power_using_impedence_model

**Comment by softwareengineerprogrammer** [comment]
Note:
1.  This suggestion primarily applies to new code only, in general we should not be refactoring and renaming existing variables/classes/methods
2. PEP standard for class names is CamelCase (not snake_case): https://peps.python.org/pep-0008/#class-names

**Comment by softwareengineerprogrammer** [comment]
Another tip - when renaming/refactoring variables in PyCharm, select the name, right click, and select Refactor -> Rename. This will ensure that any references to the name in other files are also renamed, saving work and preventing breakage.

![Screenshot 2024-01-16 at 10 19 02](https://github.com/NREL/GEOPHIRES-X/assets/4056124/996537d6-0928-44b2-8599-745a81da64df)

(Once you are familiar with this functionality, strongly recommend defining a convenient keyboard shortcut and using that instead of clicking for the sake of efficiency.)

**Comment by malcolm-dsider** [comment]
Yep, I found and used that tool a great deal during my last effort to objective SurfacePlants – sometimes, the tool will even suggest the PEP standard name for you.

 

The PEP standard is a bit confusing – methods and variables get snake case (the_is_the_variable_name) while other things, like classes, get camel case (ClassName not class_name). It took me a while to figure that out, but I think the code in the HIP-RAX branch now follows those rules for the code that I had to touch. I didn’t change many other classes.
